### PR TITLE
fix(dropdown): implement outside click hook

### PR DESCRIPTION
**Pull request by bessu4ka**
## Ticket
Link to ticket:
If someone had a similar task, then simply reject the project and delete the thread, thanks

## Description
Custom dropdown did not close when we clicked outside the component, this is now fixed

## Screenshots
image

## Checklist
- [ ] Did you test it?
- [ ] Did you add a screenshot?
- [ ] Did you fix the merge conflict?
- [ ] Did you fill the PR template?
- [ ] Did you review this PR on your own?

#### Last step if the ABOVE STEPS are COMPLETED!
- [ ] Did you share the link in a group chat?

### CODE REVIEW

File: app/modules/core/hooks/use-outside-click.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { useRef, useEffect } from 'react';
+
+export const useOutsideClick = (callback: () => void) => {
```
Comment (dev1B) [inline app/modules/core/hooks/use-outside-click.ts:3]: Could you add the option `ref` param 
Like https://chakra-ui.com/docs/hooks/use-outside-click

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,2 @@
+[*]
```
Comment (dev1B) [inline .editorconfig:1]: Could you remove this file, then add into gitignore

File: app/modules/core/hooks/use-outside-click.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  const ref = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (ref.current && !ref.current.contains(event.target as Node)) {
```
Comment (dev1B) [inline app/modules/core/hooks/use-outside-click.ts:8]: `!ref.current?.contains(...)`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#optional_chaining_with_function_calls

**Review by dev1B — CHANGES_REQUESTED** [review]

File: app/modules/core/hooks/use-outside-click.ts (hunk 1 above)
Comment (bessu4ka) [reply app/modules/core/hooks/use-outside-click.ts:3]: hmm, do you mean getting a ref like a callback?

Comment (bessu4ka) [reply app/modules/core/hooks/use-outside-click.ts:3]: Sorry, I read that the ref should be optional

**Pull request merged**